packages/wallet-connect/src/lib/wallet-connect.ts

```diff
@@ -123,10 +124,22 @@ const WalletConnect: WalletBehaviourFactory<
     throw new Error("Invalid chain id");
   };
 
-  const getAccounts = () => {
-    return (_state.session?.namespaces["near"].accounts || []).map((x) => ({
-      accountId: x.split(":")[2],
-    }));
+  const getAccounts = async (): Promise<Array<Account>> => {
+    const accounts = _state.session?.namespaces["near"].accounts || [];
+    const newAccounts = [];
+
+    for (let i = 0; i < accounts.length; i++) {
+      const keyPair = await _state.keystore.getKey(
+        options.network.networkId,
+        accounts[i].split(":")[2]
+      );
+      newAccounts.push({
+        accountId: accounts[i].split(":")[2],
+        publicKey: keyPair.getPublicKey().toString(),
```

Comment:
When I try to signIn with WalletConnect I get this error message in the modal: 
**Failed to sign in with WalletConnect: Cannot read properties of null (reading 'getPublicKey')**